types/block.go

```diff
@@ -1115,9 +1111,6 @@ func DataFromProto(dp *cmtproto.Data) (Data, error) {
 		data.Txs = Txs{}
 	}
 
-	data.SquareSize = dp.SquareSize
-	data.hash = dp.Hash
```

Comment:
this is what I was referring to @SweeXordious

if we don't do this, then when we call `data.Hash()`

https://github.com/celestiaorg/celestia-core/blob/d02553fdba2720f0314f2f6451a5d50b6755e62c/types/block.go#L1025-L1037

we will not use the celestia-app version of the hash

Reply:
this change is ignoring the hash we're passing back and instead we're using the standard tendermint one which is why the tests pass fine (that's actually the only reason we allow this to occur at all)

Reply:
it certainly doesn't have to be in this PR, but since this mechanism is so hidden it would probably be a good idea to include a test to ensure that we don't miss this in the future. Currently, those tests are only in the application, but we should be able to modify one of the test applications here to modify the data root

Reply:
https://github.com/celestiaorg/celestia-core/issues/1053

Reply:
Good catch :pray: 

what do you think about this? [f84fa8a](https://github.com/celestiaorg/celestia-core/pull/1051/commits/f84fa8abdfc659bf9119fe1e02bde35ff9c08c92)

We could either do this, or move those two lines that set the data hash and square size to the `DataFromProto` method. However, we will also need to change:

https://github.com/celestiaorg/celestia-core/blob/842a0b5ab410b200f1a1b12334a203e8671f862b/state/execution.go#L154-L156

to take the whole `rpp.Txs` slice with the square size and the data hash appended at the end.

What do you think?

Reply:
we call data.Hash() in other places, therefore we are unable to only set the value in the header

if we change one of the test applications to also modify the header, then we will be able to test this issue